pkg/ruler/rules/store.go

```diff
@@ -19,16 +19,19 @@ var (
 	ErrUserNotFound = errors.New("no rule groups found for user")
 )
 
-// RuleStore is used to store and retrieve rules
+// RuleStore is used to store and retrieve rules.
+// Methods starting with "List" prefix may return partially loaded groups: with only group Name, Namespace and User fields set.
+// To make sure that rules within each group are loaded, client must use LoadRuleGroups method.
 type RuleStore interface {
 	ListAllUsers(ctx context.Context) ([]string, error)
+	ListAllRuleGroups(ctx context.Context) (map[string]RuleGroupList, error)
+	// ListRuleGroupsForUserAndNamespace returns all the active rule groups for a user from given namespace.
+	// If namespace is empty, groups from all namespaces are returned.
+	ListRuleGroupsForUserAndNamespace(ctx context.Context, userID string, namespace string) (RuleGroupList, error)
 
-	// Returns all rule groups, and loads rules for each group.
-	LoadAllRuleGroups(ctx context.Context) (map[string]RuleGroupList, error)
+	// LoadRuleGroups loads rules for each rule group in the map.
```

Comment:
I would clarify that the `groupsToLoad` **must** be the output of either `ListAllRuleGroups` or `ListRuleGroupsForUserAndNamespace`. This comment is required because the internal implemention expects it (some `LoadRuleGroups()` implementations are no-op because groups are already loaded when listing).

Reply:
Very good point, I'll add it.